pkg/api/quota_updater.go

```diff
@@ -452,6 +513,62 @@ func (u QuotaUpdater) validateAuthorization(srv limes.ServiceInfo, oldQuota, new
 	}
 }
 
+type checkPolicyInput struct {
+	TargetDomainReport  limes.DomainReport  `json:"targetdomainreport"`
+	TargetProjectReport limes.ProjectReport `json:"targetprojectreport"`
+}
+
+// checkPolicy checks the input data against all OPA policies.
+func (u QuotaUpdater) checkPolicy(input checkPolicyInput) ([]map[string]string, *core.QuotaValidationError) {
+	results, err := u.Cluster.Config.OPAQuery.Eval(context.Background(), rego.EvalInput(input))
+	if err != nil {
+		return nil, &core.QuotaValidationError{
+			Status:  http.StatusInternalServerError,
+			Message: err.Error(),
+		}
+	}
+	if len(results) != 1 {
+		return nil, &core.QuotaValidationError{
+			Status:  http.StatusInternalServerError,
+			Message: "OPA returned unexpected amount of results (please report this problem!)",
+		}
+	}
+
+	if resultViolations, ok := results[0].Bindings["violations"].([]interface{}); ok {
+		var violations []map[string]string
+		for _, resultViolationEntry := range resultViolations {
+			if resultViolation, ok := resultViolationEntry.(map[string]interface{}); ok {
+				if _, ok := resultViolation["msg"].(string); ok {
+					violations = append(violations, map[string]string{
+						"msg":      resultViolation["msg"].(string),
+						"service":  resultViolation["service"].(string),
+						"resource": resultViolation["resource"].(string),
+					})
+				} else {
+					logg.Error("%#v", resultViolation["msg"])
+					return nil, &core.QuotaValidationError{
+						Status:  http.StatusInternalServerError,
+						Message: "OPA violation msg has wrong format (please report this problem!)",
+					}
+				}
+			} else {
+				logg.Error("OPA violation has wrong format: %s", reflect.TypeOf(resultViolation))
```

Comment:
You can simplify this with the `%T` format directive:
```suggestion
				logg.Error("OPA violation has wrong format: %T", resultViolation)
```